app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/RefreshController.kt

```diff
@@ -42,9 +42,15 @@ import kotlin.math.max
 
 object RefreshController {
 
+    private const val FORCE_CANCEL = "this_is_a_forced_cancellation_of_the_threads_algorithm"
+
     private var refreshThreadsJob: Job? = null
 
     //region Fetch Messages
+    fun cancelRefresh() {
+        refreshThreadsJob?.cancel(CancellationException(FORCE_CANCEL))
```

Comment:
Rather than using a weird string, can't you create your own Exception that extends CancellationException and pass it instead in order to detect it when you catch it ?